pkg/watch/health.go

```diff
@@ -64,17 +74,23 @@ func MonitorPodHealth(config *preparer.PreparerConfig, logger *logging.Logger, s
 			WithField("inner_err", err).
 			Fatalf("error creating health monitor KV store")
 	}
+	client, err := config.GetClient()
+	if err != nil {
+		logger.
+			WithField("inner_err", err).
+			Warningln("failed to get http client for this preparer")
```

Comment:
I noticed you changed from a `Fatalln()` to a `Warningln()`. Under what conditions can `GetClient()` fail here?